src/main/java/org/jabref/logic/shared/DBMSProcessor.java

```diff
@@ -431,6 +431,54 @@ public void removeEntry(BibEntry bibEntry) {
         return getSharedEntryList(0);
     }
 
+    public List<BibEntry> getSharedEntriesByIdList(List<Integer> idList) {
+        List<BibEntry> sharedEntries = new ArrayList<>();
+
+        StringBuilder query = new StringBuilder();
+        query.append("SELECT ")
+                .append(escape("ENTRY") + "." + escape("SHARED_ID")).append(", ")
+                .append(escape("ENTRY") + "." + escape("TYPE")).append(", ")
+                .append(escape("ENTRY") + "." + escape("VERSION")).append(", ")
+                .append("F." + escape("ENTRY_SHARED_ID")).append(", ")
+                .append("F." + escape("NAME")).append(", ")
+                .append("F." + escape("VALUE"))
+                .append(" FROM ")
+                .append(escape("ENTRY"))
+                .append(" inner join ")
+                .append(escape("FIELD"))
+                .append(" F on ")
+                .append(escape("ENTRY") + ".")
+                .append(escape("SHARED_ID") + " = F." + escape("ENTRY_SHARED_ID"))
+                .append(" where ")
+                .append(escape("SHARED_ID")).append(" in (");
+
+        for (int i = 0; i < idList.size() - 1; i++) {
```

Comment:
Somewhat nicer and easier to read:
```suggestion
        idList.stream().collect(Collectors.joining(", "))
```